.github/workflows/protobuf.yaml

```diff
@@ -0,0 +1,47 @@
+name: protobuf
+permissions: read-all
+
+on:
+  pull_request:
+    branches:
+    paths:
+      - '.github/workflows/protobuf.yaml'
+      - 'protobuf/**'
+  push:
+    branches: [ master ]
+    paths:
+      - '.github/workflows/protobuf.yaml'
+      - 'protobuf/**'
+  schedule:
+    - cron: '0 0 * * 0' # weekly
+
+jobs:
+  validate-protobuf:
+    defaults:
+      run:
+        working-directory: protobuf
+    strategy:
+      matrix:
+        sdk: [stable, dev]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        include:
+          - sdk: stable
+            run-tests: true
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
+      - uses: dart-lang/setup-dart@e51d8e571e22473a2ddebf0ef8a2123f0ab2c02c
+        with:
+          sdk: ${{matrix.sdk}}
+
+      - run: dart pub get
+
+      - run: dart analyze --fatal-infos
+
+      - run: dart format --output=none --set-exit-if-changed .
+        if: ${{matrix.run-tests && (success() || failure()) }}
```

Comment:
Is `matrix.run-tests` ever `false`? It looks hard-coded above.

Reply:
`run-tests` is true only for the `stable` sdk, so we don't test on `dev`. Feel free to change that if needed.